### DISCUSSION TITLE
env: node: No such file or directory

### DISCUSSION THREAD

**Discussion by 0xDmtri** [oven-sh/bun#10351] [Q&A]
Hi,

I install bun and then install a global package wscat via ```bun install -g wscat```.
I can see ```wscat``` binary available, but when i call ```wscat --help``` I get: ```env: node: No such file or directory```.
Why is it even searching for node and how can I fix it? Thanks!

**Comment by mhenrixon** [comment]
I had a similar issue today and solved it with `bunx --bun tailwindcss` https://discord.com/channels/486935104384532500/1356878735793717300

You might find this thread relevant.

**Comment by thelovekesh** [comment]
Issue is due to node shebang used by wcat binary 

https://github.com/websockets/wscat/blob/2cd78192a670a4e36be0aeee0ec32a4243392518/bin/wscat#L1

Either install node or run it within the bun process as suggested above.

**Comment by chase** [comment]
Another solution is to patch the package to use `bun` instead using `bun patch`. I do that for `vite` because I often forget to use `--bun` and the `.toml` option to always use `bun` doesn't seem to work.

**Comment by bb1** [comment]
is it not possible to create a symlink from node to bun?

**Comment by JaronMatzinger** [comment]
I just created a symlink to bun and named it node: `ln -s $(which bun) ~/.bun/bin/node`

**Comment by Abdeltoto** [comment]
**TL;DR** — `wscat` is a **Node/npm-style CLI**. The `wscat` launcher that got installed is almost certainly calling `env node …`. Bun installed the package, but **there is no `node` binary in your PATH**, so the kernel prints exactly: `env: node: No such file or directory`.

**Why Bun still “needs” Node here**  
`bun install -g` doesn’t magically rewrite every published CLI into a pure Bun script. Many globals still ship a `#!/usr/bin/env node` shebang — that’s normal.

**What you can do (pick one)**

1. **Easiest** — install Node (or use `nvm`/`fnm`) so `node` exists on your `PATH`, then `wscat --help` works as expected.

2. **Stay Bun-only for this tool** — skip the global and run when needed:
   ```bash
   bunx wscat --help
   ```
   (pulls/uses the package without relying on the broken global shim.)

3. **Double-check what you actually run** — `which wscat` / `head -1 $(which wscat)` — you’ll usually see the `env node` shebang. That confirms the diagnosis.

Nothing is “wrong” with Bun here — it’s just how that particular package is packaged. If you want, paste the first line of your `wscat` launcher and we can confirm 100 %.
